students/mattclau/lesson07/html_render.py

```diff
@@ -0,0 +1,160 @@
+'Main class to hold html elements'
+class Element:
+
+    'class attribute to store html tags'
+    tag = 'html'
+
+    indent = '\t'
+
+    def __init__(self, content=None, **kwargs):
+        'initialize Element with the content to be stored as a list, default list starts with the content passed in'
+        if content is None:
+            self.content = []
+        else:
+            self.content = [content]
+        if kwargs is not None:
+            self.attribs = kwargs
+        else: self.attribs = {}
+
+
+    def append(self, new_text):
+        'append method that appends new text to the element class'
+        self.content.append(new_text)
+
+    def add_attributes(self):
+
+        attributes = ''
+
+        for key, value in self.attribs.items():
+            attributes += key
+            attributes += '="'
+            attributes += value +'"'
+
+        return attributes
+
+
+    def render(self, out_file, cur_ind = ''):
+        'renders the content of the element to a file with appropriate tags'
+
+        #add attributes to go inside tag if they exist
+        if self.attribs != {}:
+            out_file.write(f'{cur_ind}<{self.tag} {self.add_attributes()}>\n')
+        else:
+            out_file.write(f'{cur_ind}<{self.tag}>\n')
```

Comment:
shouldn't need the if/else since self.add_attributes would return an empty string on empty attrs